libfreerdp/core/nla.c

```diff
@@ -1936,110 +1899,66 @@ SECURITY_STATUS nla_decrypt_public_key_hash(rdpNla* nla)
 	return status;
 }
 
-BOOL nla_read_ts_password_creds(rdpNla* nla, wStream* s)
+static BOOL nla_read_ts_credentials(rdpNla* nla, SecBuffer* data, size_t offset)
 {
-	size_t length;
-	size_t userLen = 0;
-	size_t domainLen = 0;
-	size_t passwordLen = 0;
-	const WCHAR* user = NULL;
-	const WCHAR* domain = NULL;
-	const WCHAR* password = NULL;
+	WinPrAsn1Decoder dec, dec2;
+	WinPrAsn1_OctetString credentials;
+	BOOL error;
+	WinPrAsn1_INTEGER credType;
 
 	WINPR_ASSERT(nla);
-	WINPR_ASSERT(s);
-
-	if (!nla->identity)
-	{
-		WLog_ERR(TAG, "nla->identity is NULL!");
-		return FALSE;
-	}
-
-	/* TSPasswordCreds (SEQUENCE)
-	 * Initialise to default values. */
-
-	sspi_FreeAuthIdentity(nla->identity);
-
-	if (!ber_read_sequence_tag(s, &length))
-		return FALSE;
-
-	/* The sequence is empty, return early,
-	 * TSPasswordCreds (SEQUENCE) is optional. */
-	if (length == 0)
-		return TRUE;
-
-	/* [0] domainName (OCTET STRING) */
-	if (!ber_read_contextual_tag(s, 0, &length, TRUE) || !ber_read_octet_string_tag(s, &length))
-	{
+	if (!data || data->cbBuffer < offset)
```

Comment:
Hum, is it asserts here ? I guess that if we receive such input it's a problem from the caller or I'm missing something ?

Reply:
I don't know, it was that way before but I agree that `data` should probably be asserted; I guess `data->cbBuffer < offset` shouldn't either be able to happen because decryption would fail